plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -53,6 +54,10 @@
     public static final AffixSetting<SecureString> KEY_SETTING = Setting.affixKeySetting(AZURE_CLIENT_PREFIX_KEY, "key",
         key -> SecureSetting.secureString(key, null));
 
+    /** Azure SAS token */
+    public static final AffixSetting<SecureString> SAS_TOKEN_SETTING = Setting.affixKeySetting(AZURE_CLIENT_PREFIX_KEY, "sas_token",
```

Comment:
can you add docs for the new setting as well?

Reply:
Done in https://github.com/elastic/elasticsearch/pull/42982/commits/b7045d09631d23e8a2f4c8554d382f72768d8aac :)